### PR TITLE
Handle localhost in API endpoint by using docker internal host address

### PR DESCRIPTION
**Pull request by joshmeek**
PR opened against the config update branch. Allows containers launched by the docker agent to reach out to the host machine.

### CODE REVIEW

File: src/prefect/agent/docker/agent.py [hunk 1]
```diff
@@ -381,8 +381,13 @@ def populate_env_vars(self, flow_run: GraphQLResult) -> dict:
         Returns:
             - dict: a dictionary representing the populated environment variables
         """
+        if "localhost" in config.cloud.api:
```
Comment (cicdw) [inline src/prefect/agent/docker/agent.py]: Would it be more robust to instead check
```python
if config.backend == "server":
```
?

Comment (joshmeek) [reply src/prefect/agent/docker/agent.py]: This is in case the IP set for the server backend is not localhost

Comment (cicdw) [reply src/prefect/agent/docker/agent.py]: Yea, at some point a power user might bind it to a different IP / name

Comment (cicdw) [reply src/prefect/agent/docker/agent.py]: It's also more explicit

**Pull request merged**